app/src/main/res/layout/fragment_xkpwgen.xml

```diff
@@ -75,7 +75,29 @@
             android:entries="@array/capitalization_type_values"
             android:entryValues="@array/capitalization_type_values"
             android:spinnerMode="dropdown"
+            app:layout_constraintHorizontal_weight="0.5"
             app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toBottomOf="@id/separator" />
+            app:layout_constraintTop_toBottomOf="@id/total_words" />
+
+        <com.google.android.material.textfield.TextInputLayout
+            android:id="@+id/xk_numbers_symbols_label"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="8dp"
```

Comment:
Align the start and the end of the `@id/xk_numbers_symbols_label` with the `@id/separator` and remove `marginStart`. Also, align the `Spinner` with `@id/xkPasswordText`.

Reply:
@Skrilltrax: for `xk_numbers_symbols_label` `start` can be aligned to `@id/separator` but the end in my understanding should be `app:layout_constraintEnd_toEndOf="parent"`, no? i.e. the same value as the field above.

```xml
<com.google.android.material.textfield.TextInputLayout
    android:id="@+id/xk_numbers_symbols_label"
    app:layout_constraintStart_toStartOf="@id/separator"
    app:layout_constraintEnd_toEndOf="parent"
```

and you actually meant to align the Spinner with `@+id/total_words`, no? (not with `@id/xkPasswordText` which is higher up).

Also, since the alignment is now based on the widgets above I have removed `app:layout_constraintHorizontal_weight="0.5"` on both the spinner and `xk_numbers_symbols_label`.

I think I got it right now; please verify.

Reply:
Hey @glowinthedark you're right, I meant `@+id/total_words` and not `@id/xkPasswordText`. For `xk_numbers_symbols_label` end can also be set to `@id/separator`, doing this will make sure that if we increase or decrease the start or end margin of `@id/separator` in future, then `xk_numbers_symbols_label` will automatically align with it.

Reply:
A small nit, can you set `app:layout_constraintBottom_toBottomOf="@id/xk_numbers_symbols_label"` to the `Spinner`. That should center align `Spinner` with `xk_numbers_symbols_label`